spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/dsl/task/TaskSchedule.java

```diff
@@ -14,15 +14,34 @@
  * limitations under the License.
  */
 
-package org.springframework.cloud.dataflow.integration.test.util.task.dsl;
+package org.springframework.cloud.dataflow.rest.client.dsl.task;
 
 import java.util.Arrays;
 import java.util.Map;
 
+import org.springframework.cloud.dataflow.rest.client.DataFlowOperations;
 import org.springframework.cloud.dataflow.rest.client.SchedulerOperations;
 import org.springframework.util.Assert;
 
 /**
+ * Represents a Scheduled Task defined on DataFlow server. Tasks can be scheduled defined with the help of a fluent
+ * style builder pattern or use it to retrieve and manage existing task schedules.
+ *
+ * For for instance to define a new task task schedule:
+ * <pre>
+ *     {@code
+ *      TaskScheduleBuilder taskScheduleBuilder = TaskSchedule.builder(dataFlowOperations);
+ *     	try (Task task = Task.builder(dataFlowOperations).name("myTask").definition("timestamp").create();
+ *           TaskSchedule taskSchedule = taskScheduleBuilder.prefix("mySchedule").task(task).create()) {
+ *
+ *           taskSchedule.schedule(Collections.singletonMap("scheduler.cron.expression", "56 20 ? * *"));
+ *
+ *           TaskSchedule retrievedSchedule = taskScheduleBuilder.findByScheduleName(taskSchedule.getScheduleName());
+ *           taskSchedule.unschedule();
+ *       }
+ *     }
+ * </pre>
+ *
  * @author Christian Tzolov
  */
 public class TaskSchedule implements AutoCloseable {
```

Comment:
Do we want to expose this constructor as public especially if we don't have a default version of the SchedulerOperations.  
Probably want to just have the static SchedulerBuilder to handle this scenario.

Reply:
Agree. Same should apply to the Task public constructor too.